src/stringSimilarity.js

```diff
@@ -1,17 +1,103 @@
 import escapeRegExp from 'lodash.escaperegexp';
 import matcher from 'matcher';
+import { isObject } from './helpers';
 
 export const WILDCARD_MARKER_ESCAPED = '{{\\*}}';
 export const WILDCARD_MARKER = '{{*}}';
+export const UNQUOTED_WILDCARD_PLACEHOLDER = '{{%}}';
+const searchUnquotedWildcards = new RegExp(/(([^:\s,]*)\{\{\*\}\}([^,:\s}\]]*))(?=[^,:}\]\s]*)/g);
+const searchModifiedWildcards = new RegExp(/(([^:\s,]*)\{\{%\}\}([^,\s}\]]*))(?=[^,}\]\s]*)/g);
 
-export default (source, target) => {
-  if (!source || (source || '') === (target || '')) {
-    return source === target;
-  }
-
+const getIsMatch = (source, target) => {
   const wildcardedSource = source
     .replace(new RegExp(escapeRegExp('*'), 'g'), '\\*')
     .replace(new RegExp(escapeRegExp(WILDCARD_MARKER_ESCAPED), 'g'), '*');
 
   return matcher.isMatch(target, wildcardedSource);
 };
+
+export const recursiveKeySort = (data) => {
+  if (data && isObject(data)) {
+    const sortedData = {};
+    Object.keys(data).sort().forEach((key) => {
+      sortedData[key] = recursiveKeySort(data[key]);
+    });
+    if (Array.isArray(data)) {
+      return Object.values(sortedData);
+    }
+    return sortedData;
+  }
+  return data;
+};
+
+export const replaceWildcards = (value, makeParsable) => {
+  let processedValue = value;
+  const regEx = makeParsable ? searchUnquotedWildcards : searchModifiedWildcards;
+  let resultArray = regEx.exec(processedValue);
+
+  while (resultArray !== null) {
+    const matchedString = resultArray[0];
+    const matchedStringIndex = regEx.lastIndex - matchedString.length;
+    let modifiedMatch;
+
+    if (makeParsable) {
+      const isWildcardInString = matchedString.startsWith('"')
+        && matchedString.endsWith('"');
+
+      if (!isWildcardInString) {
+        modifiedMatch = matchedString
+          .replace(WILDCARD_MARKER, UNQUOTED_WILDCARD_PLACEHOLDER);
```

Comment:
I don't think `modifiedMatch` is used outside of the if, so you could have a different name for each clause of the `if` statement. Something like `matchWithPlaceholderAndQuotes` here and `matchWithWildcardAndNoQuotes` below. You may be able to come up with a better name. Or replace lines 48-50 with a function and have the function name describe what you are doing. Same with lines 57-59. What do you think?

Reply:
Done!  Let me know if that improves readability for you.